stdlib/public/core/StringBridge.swift

```diff
@@ -225,6 +236,18 @@ internal func _bridgeCocoaString(_ cocoaString: _CocoaString) -> _StringGuts {
     return _StringGuts(_SmallString(taggedCocoa: cocoaString))
 #endif
   case .cocoa:
+    
+    let length = _stdlib_binary_CFStringGetLength(cocoaString)
+    
+    if length <= _SmallString.capacity {
+      //We *may* not be able to form a SmallString from this, but above 15 we
+      //definitely can't. It's worth trying because eager bridging is almost
+      //always good if we don't have to allocate for it.
+      if let smol = _SmallString(nonTaggedCocoa: cocoaString, length: length) {
```

Comment:
One un-addressed concern in this patch: the comments further down mention NSStrings with "associated information", and I can't for the life of me remember which possible headache that's referring to. Asking around to refresh my memory, but it could cause problems for this.

Reply:
I suspect this is "a custom mutable subclass of NSString, which might have more state than just the string data" (like, say, NSTextStorage's funky view onto its attributed contents).